spec/unleash/configuration_spec.rb

```diff
@@ -107,23 +107,35 @@
       Unleash.configure do |config|
         config.url      = 'http://test-url/'
         config.app_name = 'my-test-app'
-        config.custom_http_headers = { 'X-API-KEY': '123' }
+        config.custom_http_headers = { 'X-API-KEY' => '123' }
```

Comment:
Would prefer to keep any previous test unchanged, to the degree that it is possible. This way we can be more secure that there are no known side effects on code changes.

In a separate note, I do like using symbols better than strings for the keys, but that's secondary. Both should work.

Reply:
This is actually a great comment and I think if this test has to change to make this PR work then we probably need to talk a little bit more about details there. Mostly because there's already config in the wild and we should strive to not break that contract if possible

Reply:
We changed this because the Hash that is created in `Configuration#http_headers` is using strings as keys. All examples and tests use symbol notation which creates a mix of strings and symbols in the resulting object.

Mixing strings and symbols as keys makes the result confusing. So we wanted to avoid using symbols as keys in the existing tests.

```ruby
config = Unleash::Configuration.new(app_name: 'static', custom_http_headers: { 'UNLEASH-APPNAME': 'dynamic' })

config.http_headers
# => {"UNLEASH-INSTANCEID"=>"0c52c178-dafd-4b56-b0ce-424ce9b58848", "UNLEASH-APPNAME"=>"static", :"UNLEASH-APPNAME"=>"dynamic"}

config.http_headers["UNLEASH-APPNAME"]
# => "static"

config.http_headers[:"UNLEASH-APPNAME"]
# => "dynamic"

config.http_headers.keys.map(&:to_s)
# => ["UNLEASH-INSTANCEID", "UNLEASH-APPNAME", "UNLEASH-APPNAME"]
```

Our recommendation would be to change the tests and examples to use the same notation. We can keep the old tests to ensure that existing configurations still work.

We can open a separate PR to change this though. What do you think?

Reply:
Strings vs Symbols as keys is a known long discussion. Strings and Symbols will however both work interchangeably, and symbols have a very tiny performance edge in large hashes. (Which is not likely ever to be the case).

If one were to do something about, I think the better approach would be to convert the code and examples to use symbols as keys, instead of Strings. 

But to be honest, I think it is quite reasonable to leave things as is for now. However if you feel compelled to suggest otherwise, I agree we should do it in a separate PR.